hal/inc/cellular_hal.h

```diff
@@ -122,6 +124,7 @@ cellular_result_t cellular_credentials_set(const char* apn, const char* username
  */
 CellularCredentials* cellular_credentials_get(void* reserved);
 
+bool cellular_sim_ready(void);
```

Comment:
`bool cellular_sim_ready(void* reserved);` and we need to make sure `cellular_sim_ready` and `cellular_init` get added to the DYNALIB once the modular PR is merged to develop.  Looks good Bryce!